libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.navigation.base.route
+
+import android.net.Uri
+import com.mapbox.geojson.Point
+
+class RouteUrl(
+    val accessToken: String,
+    val orgin: Point,
+    val waypoints: List<Point>? = null,
+    val destination: Point,
+    val user: String = PROFILE_DEFAULT_USER,
+    val profile: String = PROFILE_DRIVING,
+    val steps: Boolean = true,
+    val geometries: String = GEOMETRY_POLYLINE6,
+    val overview: String = OVERVIEW_FULL,
+    val voiceIntruction: Boolean = true,
+    val bannerIntruction: Boolean = true,
+    val roundaboutExits: Boolean = true,
+    val enableRefresh: Boolean = true
+) {
+
+    companion object {
+        const val BASE_URL = "https://api.mapbox.com"
+
+        const val BASE_URL_P0 = "directions"
```

Comment:
NIT Could we come up with better names? `P0` and `P1` are confusing and don't give too much context.